Perlang.Tests.Integration/Stdlib/ArgvTests.cs

```diff
@@ -1,13 +1,23 @@
+using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Threading;
 using Perlang.Interpreter;
+using Perlang.Stdlib;
 using Xunit;
 using static Perlang.Tests.Integration.EvalHelper;
 
 namespace Perlang.Tests.Integration.Stdlib
 {
-    public class ArgvTests
+    public class ArgvTests : IDisposable
     {
+        public ArgvTests()
+        {
+            // Motivation: The Argv [ArgumentsSetter] approach is not thread safe, and xUnit runs unit tests in parallel
+            // by default.
+            Monitor.Enter(typeof(Argv));
```

Comment:
The fact that we need to do essentially a `lock(typeof(Argv))` around our test methods which uses `Argv.pop` is really an incredibly ugly code smell. We should try to find ways around that.

Perhaps make `Argv` in "Perlang scope" be not the `Argv` class, but instead an instance of `Argv`, so that `Argv.pop` would really be calling an instance method that applies to this instance of the `PerlangInterpreter`? This would be a possibility, but this has the disadvantage of making things be more "magic" and less down-to-earth. Has to be considered carefully before we decide either way.